c10/core/DispatchKey.h

```diff
@@ -150,6 +150,7 @@ enum class DispatchKey : uint8_t {
   // There are a number of alternative modes which may want to handle before
   // autograd; for example, error checking, tracing, profiling or vmap.  They
   // go here.
+  Vmap,
```

Comment:
nit: i would add 1-2 lines comment about the key it self.

Reply:
Why call it Vmap and not BatchedTensorKey or such?